xocto/ranges.py

```diff
@@ -1034,3 +1038,34 @@ def as_finite_datetime_periods(
         finite_periods += [FiniteDatetimeRange(period.start, period.end)]
 
     return finite_periods
+
+
+def iterate_over_months(
+    period: FiniteDateRange, tz: datetime.tzinfo
```

Comment:
:panda_face: can I suggest making `tz` a named argument instead of positional?

Reply:
I've made that update.